scripts/performance-tests-kevm.sh

```diff
@@ -0,0 +1,57 @@
+#!/usr/bin/env bash
+set -euxo pipefail
+
+KEVM_VERSION='v1.0.335'
+
+SCRIPT_DIR="$(dirname "$(readlink -f "$0")")"
+
+MASTER_COMMIT="$(git rev-parse origin/main)"
+
+FEATURE_BRANCH_NAME="$(git rev-parse --abbrev-ref HEAD)"
+FEATURE_BRANCH_NAME="${FEATURE_BRANCH_NAME//\//-}"
+
+if [[ $FEATURE_BRANCH_NAME == "master" ]]; then
+  FEATURE_BRANCH_NAME="feature"
+fi
+
+# Create a temporary directory and store its name in a variable.
+TEMPD=$(mktemp -d)
+
+# Exit if the temp directory wasn't created successfully.
+if [ ! -e "$TEMPD" ]; then
+    >&2 echo "Failed to create temp directory"
+    exit 1
+fi
+
+# Make sure the temp directory gets removed on script exit.
+trap "exit 1"           HUP INT PIPE QUIT TERM
+trap 'rm -rf "$TEMPD"'  EXIT
+
+
+feature_shell() {
+  GC_DONT_GC=1 nix develop . --extra-experimental-features 'nix-command flakes' --override-input k-framework/booster-backend $SCRIPT_DIR/../ --command bash -c "$1"
+}
+
+master_shell() {
+  GC_DONT_GC=1 nix develop . --extra-experimental-features 'nix-command flakes' --override-input k-framework/booster-backend github:runtimeverification/hs-backend-booster/$MASTER_COMMIT --command bash -c "$1"
+}
+
+cd $TEMPD
+git clone --depth 1 --branch $KEVM_VERSION https://github.com/runtimeverification/evm-semantics.git
+cd evm-semantics
+git submodule update --init --recursive --depth 1 kevm-pyk/src/kevm_pyk/kproj/plugin
+
+
+
+feature_shell "make poetry && poetry run -C kevm-pyk -- kevm-dist --verbose build plugin haskell --jobs 4"
+
+feature_shell "make test-prove-pyk PYTEST_PARALLEL=8 PYTEST_ARGS='--timeout 7200 -vv --use-booster' > $SCRIPT_DIR/kevm-$KEVM_VERSION-$FEATURE_BRANCH_NAME.log"
```

Comment:
With 8-fold parallelism, I am getting very high variability in the timings. When testing this PR's branch against master, we would expect there to be no significant differences but the measured differences range from 0.8 to 1.3 on my machine. (I come to think that this is because we recompile for every test, not because of the tests)
```suggestion
feature_shell "make test-prove-pyk PYTEST_PARALLEL=4 PYTEST_ARGS='--timeout 7200 -vv --use-booster' > $SCRIPT_DIR/kevm-$KEVM_VERSION-$FEATURE_BRANCH_NAME.log"
```

Reply:
Interesting... I didn't observe this variation on my linux machine. Perhaps we can make the parallelism an optional argument with 4 as default